src/ripple/app/ledger/LedgerEntrySet.cpp

```diff
@@ -1406,6 +1422,40 @@ TER LedgerEntrySet::trustDelete (
     return terResult;
 }
 
+void LedgerEntrySet::enableDeferredCredits (bool enable)
+{
+    if (!enable)
+    {
+        mDeferredCredits.reset ();
+        return;
+    }
+
+    if (!mDeferredCredits)
+        mDeferredCredits.emplace ();
+}
```

Comment:
I think asserting to prevent the use of nested `ScopedDeferredCredits` is a great idea, for all the reasons that `std::mutex` should be preferred over `std::recursive_mutex`